### PR TITLE
Bug 1019479 - [Message]The recipient which originally display as contact name will display as phone number after viewing the recipient details of a draft. Setting correct "contacts" source for the contact recipient added from draft. r=julien

### PR DESCRIPTION
**Pull request by azasypkin**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4331edbc9419bc1553d86ac552e8f65b080cbb54

File: apps/sms/js/thread_ui.js [hunk 1]
```diff
@@ -737,15 +737,18 @@ var ThreadUI = {
       // Otherwise find them from draft recipient numbers
       draft.recipients.forEach(function(number) {
         Contacts.findByAddress(number).then(function(contacts) {
+          var recipient;
           if (contacts.length) {
-            this.recipients.add(
-              Utils.basicContact(number, contacts[0])
-            );
+            recipient = Utils.basicContact(number, contacts[0]);
```
Comment (azasypkin) [inline apps/sms/js/thread_ui.js]: I was thinking about ```Object.assign({ source: 'contacts'}, Utils.basicContact(...))``` first, but it's overkill for one property only :)

File: apps/sms/js/thread_ui.js [hunk 2]
```diff
@@ -737,15 +737,18 @@ var ThreadUI = {
...
+            recipient.source = 'contacts';
           } else {
-            this.recipients.add({
-              number: number
-            });
+            recipient = {
+              number: number,
+              source: 'manual'
```
Comment (azasypkin) [inline apps/sms/js/thread_ui.js]: It's ```manual``` by default, but added it explicitly to be more consistent with other places where add number

**Comment by julienw** [comment]
Open question:

in https://github.com/mozilla-b2g/gaia/commit/3be9b8aed46d941576d90c7a14932a1e3e8fbcdd (for Tarako) we saved as draft the whole "recipient" object of all recipients (including invalid entries BTW, which yields another issue that should be fixed in my other recipient patch). How would you feel about doing the same now?

File: apps/sms/js/thread_ui.js [hunk 3]
```diff
@@ -737,15 +737,18 @@ var ThreadUI = {
...
+            };
           }
+
+          this.recipients.add(recipient);
         }.bind(this));
       }, this);
```
Comment (julienw) [inline apps/sms/js/thread_ui.js]: we should have a Recipients.fromDraft() for this (like we have for Compose). What do you think?

**Comment by azasypkin** [comment]
> Open question:
in 3be9b8a (for Tarako) we saved as draft the whole "recipient" object of all recipients (including invalid entries BTW, which yields another issue that should be fixed in my other recipient patch). How would you feel about doing the same now?

You mean saving all info including ```source``` field? But what happens if contact is removed or changed in the meantime?

File: apps/sms/js/thread_ui.js (hunk 3 above)
Comment (azasypkin) [inline apps/sms/js/thread_ui.js]: Yeah, I think it's a good idea, maybe in follow-up! But what if we rename it to ```Recipient.fromNumber()``` (that requests ```Contacts``` and validate number if needed) and we can use it not only in the ```handleDraft```?

**Comment by julienw** [comment]
Yep, this is one issue.. 

maybe when we have a local DB we'll be able to change such a stored draft when we have oncontactchange :)

File: apps/sms/js/thread_ui.js (hunk 3 above)
Comment (julienw) [inline apps/sms/js/thread_ui.js]: In my idea, the loop over numbers should be handled in Recipients too; but I also like your idea of Recipient.fromNumber().

**Comment by azasypkin** [comment]
> maybe when we have a local DB we'll be able to change such a stored draft when we have oncontactchange :)

Yeah, why not, I guess it should fire for deleted contact as well

File: apps/sms/js/thread_ui.js (hunk 3 above)
Comment (azasypkin) [inline apps/sms/js/thread_ui.js]: Ah, I see - will try different options and choose in the next patches then :)

**Pull request merged**